sycl/test/esimd/util-compiler-eval.cpp

```diff
@@ -0,0 +1,26 @@
+// RUN: %clangxx -fsyntax-only -fsycl %s
+
+// This test checks that several utility APIs can be successfully compiled.
+
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/sycl.hpp>
+
+using namespace sycl::ext::intel::experimental::esimd;
+
+enum class TheEnum { FirstMember, SecondMember, ThirdMember };
+
+static_assert(detail::is_one_of_v<float, float>, "");

```

Comment:
I believe, we shouldn't test the utility functions from 'detail' namespace in LIT tests.
@kbobrovs - FYI.

Reply:
@v-klochkov, I think testing internal utility functions in some cases can be useful. But in this particular case I agree that `detail::is_one_of_v` and `__ESIMD_EMU_DNS::is_hf_type` are not worth it. +1 for removal.